src/steps/check-google-analytics-event.ts

```diff
@@ -56,30 +56,33 @@ export class CheckGoogleAnalyticsEvent extends BaseStep implements StepInterface
                                     && r.url.includes('https://www.google-analytics.com')
                                     && r.url.includes('/collect')
                                     && r.url.includes('t=event')).map(r => r.url);
-      let actual = urls.filter(url => url.includes(`tid=${id}`)
+      const filteredRequest = urls.filter(url => url.includes(`tid=${id}`)
                                     && url.toLowerCase().includes(`ea=${eventAction.toLowerCase()}`)
                                     && url.toLowerCase().includes(`ec=${eventCategory.toLowerCase()}`));
+      let records = [];
+      if (filteredRequest.length !== 1) {
+        let table;
+        if (filteredRequest.length > 1) {
+          table = this.createTable(filteredRequest);
+          records.push(table);
+        }
+        return this.fail('Expected 1 matching GA event, but %d matched.', [filteredRequest.length], records);
+      }
+
+      let filteredRequestsWithParams;
       if (Object.keys(expectedParams).length > 0) {
-        actual = actual.filter(u => this.includesParameters(u, expectedParams));
+        filteredRequestsWithParams = filteredRequest.filter(u => this.includesParameters(u, expectedParams));
       }
 
-      if (!isNullOrUndefined(actual[0])) {
-        params = this.getUrlParams(actual[0]);
+      if (!isNullOrUndefined(filteredRequestsWithParams[0])) {
```

Comment:
- [ ] This seems to cause a `TypeError: Cannot read property '0' of undefined`, which is the test fail, but I think there are broader logical issues to resolve first.